### PR TITLE
NXP-21281 Escape prefixedPrincipalName when fetching notifs

### PR DESCRIPTION
**Pull request by egiuly-nuxeo**
https://jira.nuxeo.com/browse/NXP-21281

### CODE REVIEW

File: nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/main/java/org/nuxeo/ecm/platform/ec/notification/service/NotificationService.java [hunk 1]
```diff
@@ -548,9 +549,9 @@ public void setEmailHelper(EmailHelper emailHelper) {
 
     @Override
     public List<DocumentModel> getSubscribedDocuments(String prefixedPrincipalName) {
-        String nxql = String.format("SELECT * FROM Document WHERE ecm:mixinType ='Notifiable' "
-                + "AND ecm:isCheckedInVersion = 0 "
-                + "AND notif:notifications/*/subscribers/* = '%s'", prefixedPrincipalName);
+        String nxql = String.format(
+                "SELECT * FROM Document WHERE ecm:mixinType ='Notifiable' " + "AND ecm:isCheckedInVersion = 0 "
```
Comment (efge) [inline nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/main/java/org/nuxeo/ecm/platform/ec/notification/service/NotificationService.java]: Could you please replace the `Notifiable` part of the string with `SubscriptionAdapter.NOTIFIABLE_FACET`?

Comment (egiuly-nuxeo) [inline nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/main/java/org/nuxeo/ecm/platform/ec/notification/service/NotificationService.java]: yep, good idea!

**Comment by efge** [comment]
👍

File: nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/main/java/org/nuxeo/ecm/platform/ec/notification/service/NotificationService.java [hunk 2]
```diff
@@ -548,9 +549,9 @@ public void setEmailHelper(EmailHelper emailHelper) {
 
     @Override
     public List<DocumentModel> getSubscribedDocuments(String prefixedPrincipalName) {
-        String nxql = String.format("SELECT * FROM Document WHERE ecm:mixinType ='Notifiable' "
-                + "AND ecm:isCheckedInVersion = 0 "
-                + "AND notif:notifications/*/subscribers/* = '%s'", prefixedPrincipalName);
+        String nxql = String.format("SELECT * FROM Document WHERE ecm:mixinType = '"
```
Comment (egiuly-nuxeo) [inline nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/main/java/org/nuxeo/ecm/platform/ec/notification/service/NotificationService.java]: good point

**Pull request merged**